beacon-chain/rpc/validator/status.go

```diff
@@ -51,89 +54,99 @@ func (vs *Server) MultipleValidatorStatus(
 		if !filtered[pubkeyBytes] {
-			pubkeys = append(pubkeys, pubKey)
+			pubKeys = append(pubKeys, pubKey)
 			filtered[pubkeyBytes] = true
 		}
 	}
 	// Convert indices to public keys.
 	for _, idx := range req.Indices {
 		pubkeyBytes := headState.PubkeyAtIndex(uint64(idx))
 		if !filtered[pubkeyBytes] {
-			pubkeys = append(pubkeys, pubkeyBytes[:])
+			pubKeys = append(pubKeys, pubkeyBytes[:])
 			filtered[pubkeyBytes] = true
 		}
 	}
 	// Fetch statuses from beacon state.
-	statuses := make([]*ethpb.ValidatorStatusResponse, len(pubkeys))
-	for i, pubKey := range pubkeys {
-		statuses[i] = vs.validatorStatus(ctx, pubKey, headState)
+	statuses := make([]*ethpb.ValidatorStatusResponse, len(pubKeys))
+	indices := make([]uint64, len(pubKeys))
+	for i, pubKey := range pubKeys {
+		statuses[i], indices[i] = vs.validatorStatus(ctx, headState, pubKey)
 	}
 
 	return &ethpb.MultipleValidatorStatusResponse{
-		PublicKeys: pubkeys,
+		PublicKeys: pubKeys,
 		Statuses:   statuses,
+		Indices:    indices,
 	}, nil
 }
 
 // activationStatus returns the validator status response for the set of validators
 // requested by their pub keys.
 func (vs *Server) activationStatus(
 	ctx context.Context,
-	pubkeys [][]byte,
+	pubKeys [][]byte,
 ) (bool, []*ethpb.ValidatorActivationResponse_Status, error) {
 	headState, err := vs.HeadFetcher.HeadState(ctx)
 	if err != nil {
 		return false, nil, err
 	}
 	activeValidatorExists := false
-	statusResponses := make([]*ethpb.ValidatorActivationResponse_Status, len(pubkeys))
-	for i, key := range pubkeys {
+	statusResponses := make([]*ethpb.ValidatorActivationResponse_Status, len(pubKeys))
+	for i, pubKey := range pubKeys {
 		if ctx.Err() != nil {
 			return false, nil, ctx.Err()
 		}
-		status := vs.validatorStatus(ctx, key, headState)
-		if status == nil {
+		vStatus, idx := vs.validatorStatus(ctx, headState, pubKey)
+		if vStatus == nil {
 			continue
 		}
 		resp := &ethpb.ValidatorActivationResponse_Status{
-			Status:    status,
-			PublicKey: key,
+			Status:    vStatus,
+			PublicKey: pubKey,
+			Index:     idx,
 		}
 		statusResponses[i] = resp
-		if status.Status == ethpb.ValidatorStatus_ACTIVE {
+		if vStatus.Status == ethpb.ValidatorStatus_ACTIVE {
 			activeValidatorExists = true
 		}
 	}
 
 	return activeValidatorExists, statusResponses, nil
 }
 
-func (vs *Server) validatorStatus(ctx context.Context, pubKey []byte, headState *stateTrie.BeaconState) *ethpb.ValidatorStatusResponse {
+// validatorStatus searches for the requested validator's state and deposit to retrieve its inclusion estimate. Also returns the validators index.
+func (vs *Server) validatorStatus(
+	ctx context.Context,
+	headState *stateTrie.BeaconState,
+	pubKey []byte,
+) (*ethpb.ValidatorStatusResponse, uint64) {
 	ctx, span := trace.StartSpan(ctx, "validatorServer.validatorStatus")
 	defer span.End()
 
+	// Using farFutureEpoch as the default value for index, in case the validators index cannot be determined.
```

Comment:
It's confusing to use `farFutureEpoch` as default for index. One's unit is `Epoch` and one's unit is `ValidatorIndex`

Reply:
Agreed, is there a better way we could do this, maybe through a boolean ?

Reply:
Hmm, That would require changing the protobufs as well no? Not sure how to go about this...

Reply:
Don't name it `farFutureEpoch`, name it like:
https://github.com/prysmaticlabs/prysm/blob/master/beacon-chain/forkchoice/protoarray/types.go#L45